docs/proposals/ldap-group-sync.md

```diff
@@ -0,0 +1,105 @@
+
+```go
+// LDAPGroupDataExtractor is able to retrieve the list of members, labels, annotations and OpenShift Group
+// name for an LDAP group specified by a groupUID
+type LDAPGroupDataExtractor interface {
+  // ExtractMembers returns the list of LDAP first-class user entries that are members of the LDAP group
+  // specified by the groupUID
+	ExtractMembers(groupUID string) (members []*ldap.Entry, err error)
+  // ExtractLabels returns the set of labels that the OpenShift representation of the LDAP group specified
+  // by the groupUID should contain
+  ExtractLabels(groupUID string) (labels labels.Set, err error)
+  // ExtractAnnotations returns the mapping of annotation keys to values that the OpenShift group Representing
+  // the LDAP group specified by the groupUID should have
+  ExtractAnnotations(groupUID string) (annotations map[string]string, err error)
+  // ExtractGroupName returns the OpenShift Group name that the OpenShift Group representing the LDAP
+  // group specified by the groupUID should have
+  ExtractGroupName(groupUID string) (groupName string, err error)
+}
+```
+
+## Mapping LDAP Members to OpenShift `Identities`
+This will be done in an identical way to the mapping for [LDAP password processing](https://github.com/openshift/origin/blob/master/pkg/auth/authenticator/password/ldappassword/ldap.go#L227-243) once the requisite data is collected.
+
+```go
+// LDAPIdentityFactory creates Identites for LDAP user entries.
+type LDAPUserIdentityFactory interface {
+	IdentityFor(user *ldap.Entry) (identity userapi.Identity, err error)
+}
+```
+
+## Mapping OpenShift `Identities` Representing LDAP Members to OpenShift `Users`
+A new `UserIdentityMapper` will need to be created: the `DeterministicUserIdentityMapper`, which will assume that the unique identifier retrieved from the LDAP response and integrated into the OpenShift `Identity` will be deterministicically mapped to an OpenShift `User`. This is required as otherwise there is no deterministic mapping from the `Identity` created and any OpenShift `User`.
+
+```go
+// DeterministicUserIdentityMapper is a UserIdentityMapper that forces a deterministic mapping from
+// an Identity to a User.
+type DeterministicUserIdentityMapper struct {
+  // AllowLDAPIdentityCollisions determines if this UserIdentityMapper is allowed to make a mapping
+  // between an Identity provided by LDAP and an OpenShift User that already is mapped to another
+  // Identity provided by LDAP.
```

Comment:
"that is already mapped to another identity".  It doesn't matter where that other identity originated.